components/Utilities/SharedBus/SharedBus.c

```diff
@@ -47,27 +53,38 @@ esp_err_t SharedBusSend(SharedBusPacket_t SharedBusPacket)
         //permission to all task to peek
         EventBits = xEventGroupSetBits(
                         EventGroupHandleLocal, /* The event group being updated. */
-                        BIT_22);                               
+                        BIT_22);
+
+        return true;                               
     }
         
     return false;    
 }
+
+/**
+ * @brief Check needed Bits to receive the Packet.
+ * @param SharedBusPacket The received Packet.
+ * @param interfaceID The ID of receiver component.
+ * @return True if successfully received, false if receiver and sender are the same component.
+ */
 esp_err_t SharedBusRecieve(
     SharedBusPacket_t *SharedBusPacket,
     TaskInterfaceID_t interfaceID)
 {   
-    esp_err_t ret = false;
     EventBits = xEventGroupWaitBits(
                     EventGroupHandleLocal,   /* The event group being tested. */
                     BIT_22 | BIT_21,         /* The bits within the event group to wait for. */
-                    pdFALSE,        /* BIT_22 should NOT be cleared before returning. */
-                    pdFALSE,         /* Wait for 22th bit, either bit will do. */
-                    1);/*           Wait a maximum of 100ms for either bit to be set. */    
+                    pdFALSE,        /* BIT 21 or 22 should NOT be cleared before returning. */
+                    pdFALSE,         /* Wait for 21 or 22th bit, either bit will do. */
+                    1);/*           Wait a maximum of 1ms for either bit to be set. */    
 
-    if((EventBits & BIT_22)) //default state
-    {           
-        ret = true;    
-        if(xQueuePeek(QueueHandle, SharedBusPacket, 1) == pdTRUE)
+    if((EventBits & BIT_22) == 0) //default state
```

Comment:
I'm still waiting for an update here. You have four nested if statements, which makes the code difficult to read.